backend/src/main/kotlin/blue/mild/covid/vaxx/setup/ConfigurationDependencyInjection.kt

```diff
@@ -11,62 +15,94 @@ import org.kodein.di.singleton
             url = "jdbc:postgresql://${dbHost}/${db}"
         )
     }
 
     bind<MailJetConfigurationDto>() with singleton {
         MailJetConfigurationDto(
-            apiKey = requireNotNull(getEnv("MAIL_JET_API_SECRET")) { "MAIL_JET_API_SECRET env variable was not provided. Exiting" },
-            apiSecret = requireNotNull(getEnv("MAIL_JET_API_SECRET")) { "MAIL_JET_API_SECRET env variable was not provided. Exiting" },
-            emailFrom = getEnvOrLogDefault("MAIL_ADDRESS_FROM", "services@mild.blue"),
-            nameFrom = getEnvOrLogDefault("NAME_FROM", "Registrace Očkování")
+            apiKey = requireEnv(EnvVariables.MAIL_JET_API_KEY),
+            apiSecret = requireEnv(EnvVariables.MAIL_JET_API_SECRET),
+            emailFrom = getEnvOrLogDefault(EnvVariables.MAIL_ADDRESS_FROM, "services@mild.blue"),
+            nameFrom = getEnvOrLogDefault(EnvVariables.NAME_FROM, "Registrace Očkování")
         )
     }
 
     bind<VersionDtoOut>() with singleton { VersionDtoOut(loadVersion()) }
 
-    bind<String>("frontend") with singleton {
-        getEnvOrLogDefault("FRONTEND_PATH", "../frontend/dist/frontend")
+    bind<StaticContentConfigurationDto>() with singleton {
+        StaticContentConfigurationDto(getEnvOrLogDefault(EnvVariables.FRONTEND_PATH, "../frontend/dist/frontend"))
     }
 
-    // TODO load this from the env / config
     bind<JwtConfigurationDto>() with singleton {
         JwtConfigurationDto(
             realm = "Mild Blue Covid Vaxx",
             issuer = "vaccination.mild.blue",
             audience = "default",
-            registeredUserJwtExpirationInMinutes = 60 * 24 * 5, // 5 days
-            patientUserJwtExpirationInMinutes = 15,
-            signingSecret = UUID.randomUUID().toString()
+            registeredUserJwtExpirationInMinutes =
+            getEnvOrLogDefault(EnvVariables.JWT_EXPIRATION_REGISTERED_USER_MINUTES, "${60 * 24 * 5}").toInt(),
+            patientUserJwtExpirationInMinutes =
+            getEnvOrLogDefault(EnvVariables.JWT_EXPIRATION_PATIENT_MINUTES, "30").toInt(),
+            signingSecret =
+            getEnvOrLogDefault(EnvVariables.JWT_SIGNING_SECRET, UUID.randomUUID().toString())
         )
     }
+
+    bind<RateLimitConfigurationDto>() with singleton {
+        RateLimitConfigurationDto(
+            rateLimit =
+            getEnvOrLogDefault(EnvVariables.RATE_LIMIT, "100").toLong(),
+            rateLimitDuration =
+            getEnvOrLogDefault(EnvVariables.RATE_LIMIT_DURATION_MINUTES, "60")
+                .let { Duration.ofMinutes(it.toLong()) }
+        )
+    }
+
+    bind<SwaggerConfigurationDto>() with singleton {
+        SwaggerConfigurationDto(getEnvOrLogDefault(EnvVariables.ENABLE_SWAGGER, "true").toBoolean())
+    }
+
+    bind<CorsConfigurationDto>() with singleton {
+        val hosts = getEnvOrLogDefault(EnvVariables.CORS_ALLOWED_HOSTS, "http://localhost:4200")
+            .split(",")
+            .map { URI.create(it.trim()) }
```

Comment:
Kdyz tohle failne, tak predpokladam, ze se to ani nenastartuje.

Reply:
ano, coz je ok, mame nevalidni config